src/hotspot/share/opto/vectornode.cpp

```diff
@@ -752,7 +752,7 @@ Node* StoreVectorMaskedNode::Ideal(PhaseGVN* phase, bool can_reshape) {
       return phase->transform(new StoreVectorNode(ctr, mem, adr, adr_type(), val));
     }
   }
-  return NULL;
+  return StoreNode::Ideal(phase, can_reshape);
```

Comment:
Can you elaborate, please, why this change is needed?

Reply:
Oh, I just noticed that the `StoreVectorMaskedNode` is extended from `StoreNode`, which has some optimizations in `StoreNode::Ideal`. I'm worried that overriding the `StoreNode::Ideal` for `StoreVectorMaskedNode` might miss the optimizations in `StoreNode::Ideal`. I will reconsider this in more details. Thanks!

Reply:
It makes sense to handle it separately and fix it in the mainline.

Reply:
It's ok for me. I will revert this change. Thanks!